src/pip/_internal/commands/install.py

```diff
@@ -472,16 +473,17 @@ def run(self, options: Values, args: List[str]) -> int:
             )
             env = get_environment(lib_locations)
 
+            # Display a summary of installed packages, with extra care to
+            # display a package name as it was requested by the user.
             installed.sort(key=operator.attrgetter("name"))
             items = []
-            for result in installed:
-                item = result.name
-                try:
-                    installed_dist = env.get_distribution(item)
-                    if installed_dist is not None:
-                        item = f"{item}-{installed_dist.version}"
-                except Exception:
-                    pass
+            installed_versions = {}
+            for distribution in env.iter_all_distributions():
+                installed_versions[distribution.canonical_name] = distribution.version
+            for package in installed:
+                display_name = package.name
+                version = installed_versions.get(canonicalize_name(display_name), None)
+                item = f"{display_name}-{version}"
                 items.append(item)
```

Comment:
I'd add a new `canonical_name` attribute to `InstallationResult` to simplify this logic, but given it's not used anywhere else, this looks fine as-is. This PR is already approved so feel free to ignore this.

```python
            for package in installed:
                version = installed_versions.get(package.canonical_name, None)
                items.append(f"{package.name}-{version}")
```

Reply:
FWIW, even without adding the `canonical_name` attribute, dropping the item variable well help make this look easier to read.